core/src/scene/pointLight.cpp

```diff
@@ -40,10 +40,31 @@ void PointLight::setRadius(float _inner, float _outer){
     m_outerRadius = _outer;
 }
 
-void PointLight::setupProgram(std::shared_ptr<ShaderProgram> _shader) {
+void PointLight::setupProgram(const std::shared_ptr<View>& _view, std::shared_ptr<ShaderProgram> _shader) {
     if (m_dynamic) {
-        Light::setupProgram(_shader);
-        _shader->setUniformf(getUniformName()+".position", glm::vec4(m_position));
+        Light::setupProgram(_view, _shader);
+
+        glm::vec4 position = m_position;
+
+        if (m_origin == LightOrigin::WORLD) {
+            // For world origin, format is: [longitude, latitude, meters (default) or pixels w/px units]
+
+            // Move light's world position into camera space
+            glm::dvec2 camSpace = _view->getMapProjection().LonLatToMeters(glm::dvec2(m_position.x, m_position.y));
+            position.x = camSpace.x - _view->getPosition().x;
+            position.y = camSpace.y - _view->getPosition().y;
+            position.z = position.z - _view->getPosition().z;
+
+        } else if (m_origin == LightOrigin::GROUND) {
+            // Leave light's xy in camera space, but z needs to be moved relative to ground plane
+            position.z = position.z - _view->getPosition().z;
+        }
+        
+        if (m_origin == LightOrigin::WORLD || m_origin == LightOrigin::GROUND) {
+            position = glm::vec4(_view->getNormalMatrix() * glm::vec3(position), 1.f);
```

Comment:
I did not get why we need to multiply normal matrix to the light position here? 

It makes sense to me in the directional light case but not here ... If someone can explain :D.